app/services/search_service.rb

```diff
@@ -11,44 +11,60 @@ def initialize; end
 
   # Run a search in any of the associated systems for references that contain the search string
   # and package up as a DocResult
-  def textSearch_all(srchString)
+  def textSearch_all(search_criteria)
     sresult = DocList.new
 
     # notes
-    noteList = textSearch_notes(srchString)
+    noteList = textSearch_notes(search_criteria.query)
     sresult.addAll(noteList.items)
 
     # Node search
     Node.limit(5)
       .order('nid DESC')
-      .where('(type = "page" OR type = "place" OR type = "tool") AND node.status = 1 AND title LIKE ?', '%' + srchString + '%')
+      .where('(type = "page" OR type = "place" OR type = "tool") AND node.status = 1 AND title LIKE ?', '%' + search_criteria.query + '%')
       .select('title,type,nid,path').each do |match|
       doc = DocResult.fromSearch(match.nid, match.icon, match.path, match.title, '', 0)
       sresult.addDoc(doc)
     end
     # User profiles
-    userList = textSearch_profiles(srchString)
+    userList = textSearch_profiles(search_criteria)
     sresult.addAll(userList.items)
 
     # Tags
-    tagList = textSearch_tags(srchString)
+    tagList = textSearch_tags(search_criteria.query)
     sresult.addAll(tagList.items)
     # maps
-    mapList = textSearch_maps(srchString)
+    mapList = textSearch_maps(search_criteria.query)
     sresult.addAll(mapList.items)
     # questions
-    qList = textSearch_questions(srchString)
+    qList = textSearch_questions(search_criteria.query)
     sresult.addAll(qList.items)
 
     sresult
   end
 
-  # Search profiles for matching text and package up as a DocResult
-  def textSearch_profiles(srchString)
-    sresult = DocList.new
+  # Search profiles for matching text with optional order_by=recent param and
+  # sorted direction DESC by default
+  # then the list is packaged up as a DocResult
 
-    users = SrchScope.find_users(srchString, limit = 10) # don't return hundreds!!
-    # User profiles
+  # If no sort_by value present, then it returns a list of profiles ordered by id DESC
+  # a recent activity may be a node creation or a node revision
+  def textSearch_profiles(search_criteria)
+    user_scope = User.where('username LIKE ? AND rusers.status = 1', '%' + search_criteria.query + '%')
```

Comment:
Here i'm a bit worried -- the full text search provided by the `User.search` as part of the previous `SrchScope.find_users()` was much more efficient, and this is an open ended text match which is very expensive when run against 300k+ users. 

See:

https://github.com/publiclab/plots2/blob/master/app/services/srch_scope.rb#L5-L8

Is it possible to keep using that `.search` method so that we don't get hit hard on a very big database?

Reply:
Hey @jywarren,

We will need some help here. In my last commit, I sent the code using `SrchScope.find_users()`. It is working fine locally, but it doesn't work here... Probably because here it gets in in this condition `"if ActiveRecord::Base.connection.adapter_name == 'Mysql2'` and then calls `User.search(input)`. And locally we just run `User.where('username LIKE ? AND rusers.status = 1', '%' + input + '%')`.

Analyzing Travis' results it seems that `User.search` is not returning anything and so the error: NoMethodError: undefined method `[]' for nil:NilClass. But I don't know what we can do now, how to debug this if we are using different databases and locally it is working fine?